### PR TITLE
feat(jsii-go): use reflect to resolve overriden methods

### PR DESCRIPTION
**Pull request by yglcode**
Use reflect to resolve overridden methods, per issue #2768, leveraging
how such methods will no longer be visible on the non-pointer receiver,
while methods promoted from anonymous embeds are.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

File: packages/@jsii/go-runtime/jsii-runtime-go/runtime.go [hunk 1]
```diff
@@ -159,6 +143,36 @@ func Create(fqn FQN, args []interface{}, inst interface{}) {
 		}
 	}
 
+	//find method overrides thru reflection
+	mOverrides := getMethodOverrides(inst, "jsiiProxy_")
+	//if overriding struct has no overriding methods, could happen if
+	//overriding methods are not defined with pointer receiver.
+	if len(mOverrides) == 0 && !strings.HasPrefix(instType.Name(), "jsiiProxy_") {
+		panic(fmt.Errorf("%s has no overriding methods. Please verify overriding methods are defined with pointer receiver", instType.Name()))
```
Comment (RomainMuller) [inline packages/@jsii/go-runtime/jsii-runtime-go/runtime.go:151]:
```suggestion
		panic(fmt.Errorf("%s has no overriding methods. Overriding methods must be defined with a pointer receiver", instType.Name()))
```

**Comment by RomainMuller** [comment]
Made some (vey) minor style changes in your code (mostly adding white spaces and capitalisation in comments)...

I wonder if the error message you added could be improved a little (but I'm not too concerned about it though).

That looks pretty good to me at this stage; it's a very clever idea and I like it.

---

Would you be able to update the documentation under `gh-pages`? I can take that part up otherwise (no problem!)

**Comment by yglcode** [comment]
> Made some (vey) minor style changes in your code (mostly adding white spaces and capitalisation in comments)...
> 
> I wonder if the error message you added could be improved a little (but I'm not too concerned about it though).
> 
> That looks pretty good to me at this stage; it's a very clever idea and I like it.
> 
> Would you be able to update the documentation under `gh-pages`? I can take that part up otherwise (no problem!)

Thanks for your comments. I will update the code. It will be great if you could update the docs. Thanks.

**Comment by RomainMuller** [comment]
Updated the docs page now to remove references to the `overrides` tag and insist on the fact overridden methods must have a pointer receiver... I think we're good to ship now...

Thank you again for the clever idea & implementation!

**Review by RomainMuller — APPROVED** [review]

**Review by RomainMuller — DISMISSED** [review]

**Comment by RomainMuller** [comment]
@yglcode - don't mind me I'm trying to diagnose the transient windows PR validation failure as it appears to be happening on this particular PR (no idea why or what's wrong 🤷🏻‍♂️).

I'll get your code merged soon (even if I don't manage to understand the failure).

**Review by RomainMuller — APPROVED** [review]

**Pull request merged**